lib/Passes/ConvertXorsToCmps.cpp

```diff
@@ -146,11 +180,21 @@ ConvertXorsToCmps::run(llvm::Function &func, llvm::FunctionAnalysisManager &AM)
         continue;
       }
 
+      // invert the condition and swap the branch.
       br_inst->setCondition(var_op);
       br_inst->swapSuccessors();
+      changed = true;
     }
+
+    if (0 == ncmp_xor->getNumUses()) {
```

Comment:
Never use `getNumUses`, it has to traverse the entire use list. Instead, use `ncmp_xor->use_empty()` or `!ncmp_xor->hasNOrMoreUses(1)`